control/init.sh

```diff
@@ -22,14 +23,19 @@ if [ -z ${CI_TEST} ]; then
     if [[ $BIONIC_USE_SYSTEM_PYTHON == false ]] && hash python3.10 2>/dev/null && [[ $( source /etc/os-release; echo $VERSION_ID ) == 18.04 ]]; then
         # only append 3.10 if it is not already in the name
         if [[ $CCHQ_VIRTUALENV != *"3.10"* ]]; then
-          CCHQ_VIRTUALENV=$CCHQ_VIRTUALENV-3.10
+            OLD_VENV=$VENV
+            CCHQ_VIRTUALENV=$CCHQ_VIRTUALENV-3.10
         fi
         VENV=~/.virtualenvs/$CCHQ_VIRTUALENV
     fi
     # check if a virtualenv at $VENV exists yet, and create if not
     if [[ ! -f $VENV/bin/activate ]]; then
         if [[ $BIONIC_USE_SYSTEM_PYTHON == false ]] && hash python3.10 2>/dev/null; then
             echo "Creating a python3.10 virtual environment named ${CCHQ_VIRTUALENV}"
+            if [[ $OLD_VENV != '' ]]; then
+                echo "Your old virtual environment will remain at ${OLD_VENV}"
+                echo "If you wish to delete it, run 'rm -rf ${OLD_VENV}'"
+            fi
```

Comment:
```suggestion
            if [ -n "$CCHQ_VENV_OLD" ]; then
                echo "Your old virtual environment will remain at ${CCHQ_VENV_OLD}"
                echo "If you wish to delete it, run 'rm -rf ${CCHQ_VENV_OLD}'"
            fi
```

Reply:
Ah thanks. It looks like `[ $OLD_VENV ]` is sufficient and more similar to python so I'm going to go with that.

Reply:
See e564f4dcd3d1ed50d8af11e0a715e609170735e9

Reply:
Note that quoting the variable is recommended because otherwise we're making assumption about its content, and also that `test` and `test -n` are not exactly the same thing, and `test -n` would be more clear and explicit in this case.

Reply:
What about [these docs](https://tldp.org/LDP/Bash-Beginners-Guide/html/sect_07_01.html)?

> [ -n STRING ] or [ STRING ] | True if the length of "STRING" is non-zero.

Assuming I will wrap it in quotes, it guarantees the content will be a string so the `[ "$CCHQ_VENV_OLD" ]` check would be clear right?

Reply:
Offlined with Marco and added the `-n` option even though `[ STRING ]` and `[ -n STRING ]` do behave the same.